rclcpp/src/rclcpp/publisher_base.cpp

```diff
@@ -145,21 +145,39 @@ PublisherBase::bind_event_callbacks(
       event_callbacks.liveliness_callback,
       RCL_PUBLISHER_LIVELINESS_LOST);
   }
+
+  QOSOfferedIncompatibleQoSCallbackType incompatible_qos_cb;
   if (event_callbacks.incompatible_qos_callback) {
-    this->add_event_handler(
-      event_callbacks.incompatible_qos_callback,
-      RCL_PUBLISHER_OFFERED_INCOMPATIBLE_QOS);
+    incompatible_qos_cb = event_callbacks.incompatible_qos_callback;
+  } else if (use_default_callbacks) {
+    // Register default callback when not specified
+    incompatible_qos_cb = [this](QOSOfferedIncompatibleQoSInfo & info) {
+        this->default_incompatible_qos_callback(info);
+      };
+  }
+  try {
+    if (incompatible_qos_cb) {
+      this->add_event_handler(incompatible_qos_cb, RCL_PUBLISHER_OFFERED_INCOMPATIBLE_QOS);
+    }
+  } catch (const UnsupportedEventTypeException & /*exc*/) {
+    // pass
```

Comment:
a debug message wouldn't be a bad idea

Reply:
Good call, fixed in d21e1cae5af762ee044893fc48fe32061d23ad49.